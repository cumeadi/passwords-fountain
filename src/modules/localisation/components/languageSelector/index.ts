export { LanguageSelector } from './languageSelector.component';
