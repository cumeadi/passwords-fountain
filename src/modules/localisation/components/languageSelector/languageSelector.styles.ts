import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme } from '@/common/theme';
import {
    Wrapper as Dropdown,
    SelectedOptionWrapper,
    OptionsWrapper,
} from '@/common/components/dropdown/dropdown.styles';

setPragma(h);

export const Wrapper = styled('section')`
    display: grid;
    place-items: center;

    ${Dropdown} {
        min-width: initial;
        display: inline-grid;
    }

    ${SelectedOptionWrapper} {
        justify-items: end;
        padding: 0;
        background: transparent;
    }

    ${OptionsWrapper} {
        width: auto;
        right: ${theme.spacing.xs6};
        top: calc(100% + ${theme.spacing.xs6});
    }
`;
