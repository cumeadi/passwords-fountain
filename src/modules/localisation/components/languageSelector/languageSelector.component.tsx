import { h, VNode } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import { Wrapper } from './languageSelector.styles';
import {
    Dropdown,
    DropdownOption,
} from '@/common/components/dropdown/dropdown.component';
import { Icon } from '@/common/components/icon';
import { Text } from '@/modules/localisation/components/text';
import { languages } from '@/modules/localisation/i18n';
import { useLocalisation } from '@/modules/localisation/localisation.context';

export const LanguageSelector: TypedComponent<Props> = () => {
    const [, setLanguage] = useLocalisation();
    const options = [
        { id: 0, label: 'language.english', value: languages.en },
        { id: 1, label: 'language.polish', value: languages.pl },
    ];

    return (
        <Wrapper>
            <Dropdown
                withDefaultOption={options[0]}
                renderOptionList={(
                    OptionComponent: any,
                    onOptionClick: Function
                ): VNode[] => {
                    return options.map((option: DropdownOption) => (
                        <OptionComponent
                            key={option.id}
                            onClick={(): void => {
                                onOptionClick(option);
                                setLanguage(option.value as string);
                            }}
                        >
                            <Text>{option.label}</Text>
                        </OptionComponent>
                    ));
                }}
                renderSelectedOption={(): null => null}
                renderExpandIcon={(): VNode => (
                    <Icon name="globe" width={30} height={30} />
                )}
            />
        </Wrapper>
    );
};

interface Props {}

LanguageSelector.propTypes = {};
