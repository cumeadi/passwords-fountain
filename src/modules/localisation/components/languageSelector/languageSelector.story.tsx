import { h, VNode } from 'preact';
import { LanguageSelector } from './languageSelector.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'LanguageSelector',
    decorators: [withA11y],
};

const wrapperStyles = {
    margin: '10px 120px' /* stylelint-disable-line */,
};

export const defaultView = (): VNode => (
    <div style={wrapperStyles}>
        <LanguageSelector />
    </div>
);
