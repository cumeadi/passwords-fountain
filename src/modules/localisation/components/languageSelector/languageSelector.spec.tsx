import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { LanguageSelector } from './languageSelector.component';

describe('LanguageSelector', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<LanguageSelector />);

        expect(asFragment()).toMatchSnapshot();
    });
});
