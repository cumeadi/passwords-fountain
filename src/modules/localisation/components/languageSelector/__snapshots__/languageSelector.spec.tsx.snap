// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go2164034101"
  >
    <section
      class="go62670245"
      tabindex="0"
    >
      <div
        class="go3942427238"
      >
        <img
          alt="icon"
          class="go115679"
          height="30"
          src="static-asset-stub"
          width="30"
        />
      </div>
      <div
        class="go129576329"
      />
    </section>
  </section>
</DocumentFragment>
`;
